nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/users/ui/user-listing/user-table/user-table.component.ts

```diff
@@ -0,0 +1,245 @@
+            });
+        });
+
+        this.dataSource.data = this.sortUsers(tenantItems, this.sort);
+        this.dataSource.filterPredicate = (data: TenantItem, filter: string) => {
+            const { filterTerm, filterColumn } = JSON.parse(filter);
+            if (filterColumn === 'user') {
+                return this.nifiCommon.stringContains(data.user, filterTerm, true);
+            } else {
+                return this.nifiCommon.stringContains(this.formatMembership(data), filterTerm, true);
+            }
+        };
+        this.totalCount = tenantItems.length;
+        this.filteredCount = tenantItems.length;
+
+        // apply any filtering to the new data
+        const filterTerm = this.filterForm.get('filterTerm')?.value;
+        if (filterTerm?.length > 0) {
+            const filterColumn = this.filterForm.get('filterColumn')?.value;
+            this.applyFilter(filterTerm, filterColumn);
+        }
+    }
+
+    @Input() selectedTenantId!: string;
+    @Input() currentUser!: CurrentUser;
+    @Input() configurableUsersAndGroups!: boolean;
+
+    @Output() createTenant: EventEmitter<void> = new EventEmitter<void>();
+    @Output() selectTenant: EventEmitter<string> = new EventEmitter<string>();
+    @Output() editTenant: EventEmitter<string> = new EventEmitter<string>();
+    @Output() deleteUser: EventEmitter<UserEntity> = new EventEmitter<UserEntity>();
+    @Output() deleteUserGroup: EventEmitter<UserGroupEntity> = new EventEmitter<UserGroupEntity>();
+    @Output() viewAccessPolicies: EventEmitter<string> = new EventEmitter<string>();
+
+    sort: Sort = {
+        active: 'user',
+        direction: 'asc'
+    };
+
+    constructor(
+        private formBuilder: FormBuilder,
+        private nifiCommon: NiFiCommon
+    ) {
+        this.filterForm = this.formBuilder.group({ filterTerm: '', filterColumn: 'user' });
+    }
+
+    ngAfterViewInit(): void {
+        this.filterForm
+            .get('filterTerm')
+            ?.valueChanges.pipe(debounceTime(500))
+            .subscribe((filterTerm: string) => {
+                const filterColumn = this.filterForm.get('filterColumn')?.value;
+                this.applyFilter(filterTerm, filterColumn);
+            });
+
+        this.filterForm.get('filterColumn')?.valueChanges.subscribe((filterColumn: string) => {
+            const filterTerm = this.filterForm.get('filterTerm')?.value;
+            this.applyFilter(filterTerm, filterColumn);
+        });
+    }
+
+    applyFilter(filterTerm: string, filterColumn: string) {
+        this.dataSource.filter = JSON.stringify({ filterTerm, filterColumn });
+        this.filteredCount = this.dataSource.filteredData.length;
+    }
+
+    updateSort(sort: Sort): void {
+        this.sort = sort;
+        this.dataSource.data = this.sortUsers(this.dataSource.data, sort);
+    }
+
+    formatMembership(item: TenantItem): string {
+        return item.membership.join(', ');
```

Comment:
consider sorting the membership array before joining.
It could be that 2 groups of memberships have the same users but are in the array in different order... it would make them display differently and sort differently in the table.

```suggestion
        return item.membership.sort((a, b) => this.nifiCommon.compareString(a, b)).join(', ');


```